modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1214,6 +1233,19 @@ def apply_func_to_indices_both_axis(
         if col_widths is None:
             col_widths = [None] * len(col_partitions_list)
 
+        if row_lengths is None and col_widths is None:
```

Comment:
The `length` and `width` values of each partition are accessed in the local `compute_part_size`, defined immediately below. The double `for` loop structure where `compute_part_size` is called makes it hard to parallelize the computation of these dimensions, so I thought it would be simplest to precompute the relevant dimensions before the loop.